component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/x509Certificate/X509CertificateAuthenticator.java

```diff
@@ -325,4 +396,118 @@ private void allowUser(String userName, Map claims, X509Certificate cert,
+                    X509CertificateConstants.X509_CERTIFICATE_ALTERNATIVE_NAMES_REGEX_MULTIPLE_MATCHES_ERROR_CODE);
+            throw new AuthenticationFailedException("More than one match for the given regex");
+        } else {
+            return matchedAlternativeNamesList.get(0);
+        }
+
+    }
+
+    /**
+     * Get decoded alternative name
+     *
+     * @param decoder ASN1 Decoder
+     */
+    private String decodeAlternativeName(ASN1InputStream decoder) throws IOException {
+
+        DEREncodable encoded = decoder.readObject();
+        encoded = ((DERSequence) encoded).getObjectAt(1);
+        encoded = ((DERTaggedObject) encoded).getObject();
+        encoded = ((DERTaggedObject) encoded).getObject();
+        return  ((DERUTF8String) encoded).getString();
+    }
+
+    /**
+     * validate the certificate using the selected subject.
+     *
+     * @param subject               matched string or the username that uses to authenticate.
+     * @param authenticationContext authenticationContext.
+     * @param cert                  x509 certificate.
+     * @param claims                user claims.
+     */
+    private void validateUsingSubject(String subject, AuthenticationContext authenticationContext,
+            X509Certificate cert, Map<ClaimMapping, String> claims) throws AuthenticationFailedException {
+
+        byte[] data;
+        try {
+            data = cert.getEncoded();
+        } catch (CertificateEncodingException e) {
+            throw new AuthenticationFailedException(
+                    "Encoded certificate is not found in the certificate with subjectDN: " + cert.getSubjectDN(), e);
+        }
+        AuthenticatedUser authenticatedUser = getUsername(authenticationContext);
+
+        if (log.isDebugEnabled()) {
+            log.debug("Getting X509Certificate username");
+        }
+        
+        if (authenticatedUser != null) {
+            if (log.isDebugEnabled()) {
+                log.debug("Authenticated username is: " + authenticatedUser);
+            }
+            String authenticatedUserName = authenticatedUser.getAuthenticatedSubjectIdentifier();
+            if (authenticatedUserName.equals(subject)) {
+                addOrValidateCertificate(subject, authenticationContext, data, claims, cert);
+            } else {
+                authenticationContext.setProperty(X509CertificateConstants.X509_CERTIFICATE_ERROR_CODE,
+                        X509CertificateConstants.USERNAME_CONFLICT);
+                throw new AuthenticationFailedException(
+                        "Couldn't find X509 certificate to this authenticated user: " + authenticatedUserName);
+            }
+        } else {
+            addOrValidateCertificate(subject, authenticationContext, data, claims, cert);
+        }
+    }
+
+    private void addMatchStringsToList(Matcher matcher, List matches) {
+
+        while (matcher.find()) {
+            matches.add(matcher.group());
+        }
+    }
+
+
```

Comment:
Remove unwanted empty lines

Reply:
Fixed with this PR https://github.com/wso2-extensions/identity-outbound-auth-x509/pull/30/